app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java

```diff
@@ -324,6 +337,87 @@ public String string(@StringRes int resId) {
+     *
+     * @param fileName Path of file to be encrypted
+     */
+    private void setPassword(final String fileName) {
+        final MaterialDialog dialog = new MaterialDialog.Builder(mActivity)
+                .title(R.string.set_password)
+                .customView(R.layout.custom_dialog, true)
+                .positiveText(android.R.string.ok)
+                .negativeText(android.R.string.cancel)
+                .build();
+
+        mPositiveAction = dialog.getActionButton(DialogAction.POSITIVE);
+        mPasswordInput = dialog.getCustomView().findViewById(R.id.password);
+        mPasswordInput.addTextChangedListener(
+                new TextWatcher() {
+                    @Override
+                    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+                    }
+
+                    @Override
+                    public void onTextChanged(CharSequence s, int start, int before, int count) {
+                        mPositiveAction.setEnabled(s.toString().trim().length() > 0);
+                    }
+
+                    @Override
+                    public void afterTextChanged(Editable input) {
+                        if (StringUtils.isEmpty(input)) {
+                            Snackbar.make(Objects.requireNonNull(mActivity).findViewById(android.R.id.content),
+                                    R.string.snackbar_password_cannot_be_blank,
+                                    Snackbar.LENGTH_LONG).show();
+                        } else {
+                            mPassword = input.toString();
+
+                        }
+                    }
+                });
+        dialog.show();
+        mPositiveAction.setEnabled(false);
+        mPositiveAction.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                try {
+                    doEncryption(fileName);
+                } catch (IOException | DocumentException e) {
+                    e.printStackTrace();
+                }
+                dialog.dismiss();
+                notifyDataSetChanged();
+                Snackbar.make(Objects.requireNonNull(mActivity).findViewById(android.R.id.content),
+                        R.string.password_added,
+                        Snackbar.LENGTH_LONG).show();
+            }
+        });
+
+    }
+
+    /**
+     * Uses PDF Reader to set encryption in pdf file.
+     *
+     * @param path Path of pdf file to be encrypted
+     */
+    private void doEncryption(String path) throws IOException, DocumentException {
+        final String[] name_withpdf = path.split("/");
+        String final_path = "";
+        String filename= name_withpdf[name_withpdf.length - 1];
+        for(int i=0;i<name_withpdf.length-1;i++)
+        {
+            final_path = final_path + name_withpdf[i]+"/";
+        }
+
+        final String[] name_withoutpdf = filename.split("\\.");
```

Comment:
You have quite messed up the logic here.

Why are you splitting it by `/`
Just remove the `.pdf` from `path` & replace it with `_encrypted.pdf`